csr_check.go

```diff
@@ -128,12 +128,20 @@ func authorizeCSR(config ClusterMachineApproverConfig, machines []v1beta1.Machin
 		return authorizeNodeClientCSR(config, machines, nodes, req, csr)
 	}
 
-	// node serving cert validation after this point
+	isNodeServingCert, err := isNodeServingCert(req, csr)
+	if isNodeServingCert {
```

Comment:
Shouldn't this check `err` first? If the implementation changed to return `true, a-real-error` this would unexpectedly pass.